python/setup.py

```diff
@@ -1,11 +1,41 @@
 import os
+import codecs
 import sys
 import shutil
 import subprocess
 from pathlib import Path
 
 from setuptools import Extension, setup
 from setuptools.command.build_ext import build_ext
+import setuptools_scm as scm
+import setuptools_scm.git as scmgit
+
+def get_git_revision_short_hash() -> str:
+    try:
+        sha = subprocess.check_output(['git', 'rev-parse', '--short', 'HEAD']).decode('ascii').strip()
+        return f"+{sha}"
+    except Exception:
+        return ''
+
+def read(rel_path):
+    here = os.path.abspath(os.path.dirname(__file__))
+    with codecs.open(os.path.join(here, rel_path), 'r') as fp:
+        return fp.read()
+
+def get_version(rel_path):
+    new_ver = None
+    with open(rel_path) as f:  lines = f.read().splitlines()
+    with open(rel_path, 'w') as f:
+        for line in lines:
+            if line.startswith('__version__'):
+                delim = '"' if '"' in line else "'"
+                new_ver = line.split(delim)[1].split("+")[0] + get_git_revision_short_hash()
```

Comment:
You don't need to get `git` revision manually. It is already returned from `scmgit.parse` in `ScmVersion` instance as a `ScmVersion.node` field. You also has `ScmVersion.distance`.

Reply:
If scmgit.parse(root, config) returns None then this ScmVersion.node is also None. So If I left something like that:
```
ver = scmgit.parse(root, config)
    if (ver == '0.0') or (ver is None):
        version = get_version("hyperon/_version.py") + scmgit.ScmVersion.node
        return scm.version.meta(tag=version, config=config, preformatted=True, dirty=True)
    else:
        return ver
```
I'll get an error that  scmgit.ScmVersion.node is None since if is True if ver from scmgit.parse is None. Though probably you meant something else?

Reply:
If version is `None` it means `setuptools_scm` didn't find git repo or git command, so you will not find the distance either. Thus I would suggest return bare version if `setuptools_scm` returned None. Even simpler way is to redefined `version_scheme`